From 2db96217e7e515071726ca4ec791742c4202a1b2 Mon Sep 17 00:00:00 2001
From: Ye Bin <yebin10@huawei.com>
Date: Wed, 19 Oct 2022 11:36:02 +0800
Subject: [PATCH] blktrace: remove unnessary stop block trace in
 'blk_trace_shutdown'
Git-commit: 2db96217e7e515071726ca4ec791742c4202a1b2
Patch-mainline: v6.1-rc2
References: bsn#12
Modified-by-SEL: Yes, modified due to different context


As previous commit, 'blk_trace_cleanup' will stop block trace if
block trace's state is 'Blktrace_running'.
So remove unnessary stop block trace in 'blk_trace_shutdown'.

Signed-off-by: Ye Bin <yebin10@huawei.com>
Reviewed-by: Christoph Hellwig <hch@lst.de>
Link: https://lore.kernel.org/r/20221019033602.752383-4-yebin@huaweicloud.com
Signed-off-by: Jens Axboe <axboe@kernel.dk>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 kernel/trace/blktrace.c |    4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

--- a/kernel/trace/blktrace.c
+++ b/kernel/trace/blktrace.c
@@ -781,10 +781,8 @@ void blk_trace_shutdown(struct request_q
 {
 	mutex_lock(&q->debugfs_mutex);
 	if (rcu_dereference_protected(q->blk_trace,
-				      lockdep_is_held(&q->debugfs_mutex))) {
-		__blk_trace_startstop(q, 0);
+				      lockdep_is_held(&q->debugfs_mutex)))
 		__blk_trace_remove(q);
-	}
 
 	mutex_unlock(&q->debugfs_mutex);
 }
